modules/standard/Barrier.chpl

```diff
@@ -17,349 +17,9 @@
  * limitations under the License.
  */
 
-/* Support for task barriers.
-
-   The Barrier type provided in this module can be used to prevent tasks
-   from proceeding until all other related tasks have also reached the
-   barrier.
-
-   In the following example, all of the tasks created by the coforall loop
-   will print their "entering the barrier" messages in an unspecified order,
-   then all of the tasks will print their "past the barrier" messages, also
-   in an unspecified order.  Because of the barrier, all of the
-   "entering the barrier" messages will be printed before any of the
-   "past the barrier" messages. ::
-
-     use Barrier;
-
-     config const numTasks = here.maxTaskPar;
-     var b = new Barrier(numTasks);
-
-     coforall tid in 1..numTasks {
-       writeln("Task ", tid, " is entering the barrier");
-       b.barrier();
-       writeln("Task ", tid, " is past the barrier");
-     }
-
-   Note: When a barrier instance goes out of scope it is automatically deleted.
-   After it is deleted, any copies of the barrier that remain are invalid.
-
-   Future direction
-   ----------------
-   In the future, we expect to add more language-level support for a
-   "task-team" concept.  A task-team will more directly support collective
-   operations such as barriers between the tasks within a team.
-
-   The current implementation is designed for correctness, but is not expected
-   to perform well at scale.  We expect performance at scale to improve as
-   this barrier implementation is optimized and as the task-team concept is
-   implemented and optimized.
-*/
+/* Deprecated. Use the module :mod:`Barriers` instead. */
```

Comment:
Note that we removed the documentation of `LayoutCSR.chpl` from the public docs, so we should resolve this inconsistency in how we document deprecated modules. If this isn't a well-defined protocol (which it doesn't appear to be), then we should get input from the broader team and try to define a protocol.